src/main/java/com/imaee/propinq/neighborhood/data/models/Locality.java

```diff
@@ -0,0 +1,34 @@
+package com.imaee.propinq.neighborhood.data.models;
+
+import com.imaee.propinq.shared.data.models.Locatable;
+import jakarta.persistence.*;
+import lombok.*;
+
+import java.util.UUID;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Data
+@Table(uniqueConstraints = {
+        @UniqueConstraint(
+                name = "UniqueNameAndProvince",
+                columnNames = {
+                        "name",
+                        "province"
+                }
+        )
+})
+public class Locality extends Locatable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
```

Comment:
Cambiemos esto por lo siguiente:
```suggestion
    private final UUID localityId  = UUID.randomUUID();
```

Reply:
No sería más correcto un uso de JPA que manual?